sound/soc/sof/intel/hda-stream.c

```diff
@@ -279,6 +279,41 @@ int hda_dsp_stream_put(struct snd_sof_dev *sdev, int direction, int stream_tag)
 	return 0;
 }
 
+int hda_dsp_stream_reset(struct snd_sof_dev *sdev, struct hdac_stream *hstream)
```

Comment:
It is actually a reset toggling, not reset operation we should take at "RESET STREAM FLOW from PAUSED"?

On the other hand, I don't see a not from @plbossart 's email about the note "the proper HDA programming recommendation is to do a stream reset before coupling the host and link DMA's".

If hw_free() is called each time a stream is stopped, we should already reset the stream there, do you mean that hw_free() is not necessary called sometimes, so we have to perform an extra reset before a subsequent hw_params()?

The logic could be much clear to me if it goes like this:
1. set SRST=1 (not a 0->1->0 toggle) at initialization.
2. perform RESET->PAUSED transition at hw_params(), set SRST=0.
3. perform PAUSED->RUNNING transition at trigger_start, set RUN=1.
4. ....
5. perform RUNNING->PAUSED transition at trigger stop via IPC, set RUN=0.
6. perform PAUSED->RESET transition at the subsequent hw_free(), set SRST=0.

With this logic, we don't need explicit SRST toggling at all, no?